SPECS-EXTENDED/suitesparse/suitesparse.spec

```diff
@@ -0,0 +1,752 @@
+    popd
+    cp -p Include/*.h ../Include
+    cp -p README.txt Doc/lesser.txt ../Doc/KLU
+  popd
+
+  pushd LDL
+    pushd Lib
+      %make_build CFLAGS="$CFLAGS" BLAS="$BLAS" LIBRARY_SUFFIX="$LIBRARY_SUFFIX"
+    popd
+    cp -p Include/*.h ../Include
+    cp -p README.txt Doc/ChangeLog Doc/lesser.txt Doc/*.pdf ../Doc/LDL
+  popd
+
+  pushd UMFPACK
+    pushd Lib
+      %make_build CFLAGS="$CFLAGS" BLAS="$BLAS" LIBRARY_SUFFIX="$LIBRARY_SUFFIX"
+    popd
+    cp -p Include/*.h ../Include
+    cp -p README.txt Doc/License.txt Doc/ChangeLog Doc/gpl.txt Doc/*.pdf ../Doc/UMFPACK
+  popd
+
+  pushd SPQR
+    pushd Lib
+      %make_build CFLAGS="$CFLAGS -DHAVE_TBB -DNPARTITION" TBB=-ltbb BLAS="$BLAS" LIBRARY_SUFFIX="$LIBRARY_SUFFIX"
+    popd
+    cp -p Include/*.h* ../Include
+    cp -p README{,_SPQR}.txt
+    cp -p README_SPQR.txt Doc/* ../Doc/SPQR
+  popd
+
+  pushd RBio
+    pushd Lib
+      %make_build CFLAGS="$CFLAGS" BLAS="$BLAS" LIBRARY_SUFFIX="$LIBRARY_SUFFIX"
+    popd
+    cp -p Include/*.h ../Include
+    cp -p README.txt Doc/ChangeLog Doc/License.txt ../Doc/RBio
+  popd
+
+  popd
+done
+
+%install
+mkdir -p ${RPM_BUILD_ROOT}%{_libdir}
+mkdir -p ${RPM_BUILD_ROOT}%{_includedir}/%{name}
+cp -a SuiteSparse/Include/*.{h,hpp} ${RPM_BUILD_ROOT}%{_includedir}/%{name}/
+for build in SuiteSparse %{?build64:SuiteSparse64 SuiteSparse64_}
+do
+  pushd $build
+    cp -a */Lib/*.a lib/*.so* ${RPM_BUILD_ROOT}%{_libdir}/
+    chmod 755 ${RPM_BUILD_ROOT}%{_libdir}/*.so.*
+
+    # collect licenses in one place to ship as base package documentation
+    rm -rf Licenses
+    mkdir Licenses
+    find */ -iname lesser.txt -o -iname license.txt -o -iname gpl.txt -o \
+        -iname license | while read f; do
+            b="${f%%/*}"
+            r="${f#$b}"
+            x="$(echo "$r" | sed 's|/doc/|/|gi')"
+            install -m0644 -D "$f" "./Licenses/$b/$x"
+        done
+
+    # hardlink duplicate documentation files
+    hardlink -cv Docs/ Licenses/
+  popd
+done
+
+
+%check
+export AUTOCC=no
+export CC=gcc
+TESTDIRS="AMD CAMD CCOLAMD CHOLMOD COLAMD KLU LDL SPQR RBio UMFPACK"
+%if "%{?enable_csparse}" == "1"
+TESTDIRS="$TESTDIRS CSparse"
+%else
+TESTDIRS="$TESTDIRS CXSparse"
+%endif
+for build in SuiteSparse %{?build64:SuiteSparse64 SuiteSparse64_}
+do
+  pushd $build
+    export CFLAGS="$RPM_OPT_FLAGS -I%{_includedir}/metis"
+    export LAPACK=""
+    # Set flags for ILP64 build
+    if [ $build = SuiteSparse64 ]
+    then
+       export CFLAGS="$CFLAGS -DBLAS64"
+       export BLAS=-lopenblas64
+       export LIBRARY_SUFFIX=64
+    elif [ $build = SuiteSparse64_ ]
+    then
+       export CFLAGS="$CFLAGS -DBLAS64 -DSUN64"
+       export BLAS=-lopenblas64_
+       export LIBRARY_SUFFIX=64_
+    else
+       export BLAS=-lopenblas
+    fi   
+
+    for d in $TESTDIRS ; do
+        %make_build -C $d/Demo CFLAGS="$CFLAGS" LIB="%{?__global_ldflags} -lm -lrt" BLAS="$BLAS" LIBRARY_SUFFIX="$LIBRARY_SUFFIX" SPQR_CONFIG=-DHAVE_TBB TBB=-ltbb
```

Comment:
Ptests will show false positives if any of the tests fail, because the exit code for `%check` will be the code for the last command (end of the `for` loop or `popd` - not sure now). Let's monitor the status of the tests and if at least one failed, let's return that at the end of the `%check` section.